src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,114 @@
+            if (generatedKeys.next()) {
+                manufacturer.setId(generatedKeys.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(String.format("Can't insert manufacturer "
+                    + "to the DB %s", manufacturer.toString()), e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String getQuery = "SELECT * FROM manufacturers WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getStatement = connection.prepareStatement(getQuery)) {
+            getStatement.setLong(1, id);
+            ResultSet resultSet = getStatement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getManufacturer(resultSet));
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException(String.format("Can't get manufacturer by id %s",
+                    id), e);
+        }
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        String selectAllQuery = "SELECT * FROM manufacturers WHERE is_deleted = FALSE;";
+        List<Manufacturer> manufacturers = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                Statement getAllStatement = connection.createStatement()) {
+            ResultSet resultSet = getAllStatement.executeQuery(selectAllQuery);
+            while (resultSet.next()) {
+                manufacturers.add(getManufacturer(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all Manufacturers", e);
+        }
+        return manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String updateRequest = "UPDATE manufacturers SET name = ?, country = ? "
+                + "WHERE id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateManufacturerStatement =
+                        connection.prepareStatement(updateRequest)) {
+            updateManufacturerStatement.setString(1, manufacturer.getName());
+            updateManufacturerStatement.setString(2, manufacturer.getCountry());
+            updateManufacturerStatement.setLong(3, manufacturer.getId());
+            if (updateManufacturerStatement.executeUpdate() > 0) {
+                return manufacturer;
+            }
+            throw new DataProcessingException("Can't find manufacturer with a given id: "
+                    + manufacturer.getId() + " in the DB");
+        } catch (SQLException e) {
+            throw new DataProcessingException(String.format("Can't update manufacturer with id: %s",
+                    manufacturer.getId()), e);
+        }
```

Comment:
навіщо в try викидати помилку ?
```suggestion
            updateManufacturerStatement.executeUpdate()
            return manufacturer;
        } catch (SQLException e) {
            throw new DataProcessingException(String.format("Can't update manufacturer with id: %s",
                    manufacturer.getId()), e);
        }
```